core-java-modules/core-java-io-apis-2/src/test/java/com/baeldung/inputstream/InputStreamUnitTest.java

```diff
@@ -0,0 +1,98 @@
+package com.baeldung.inputstream;
+
+import static org.junit.jupiter.api.io.CleanupMode.ALWAYS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class InputStreamUnitTest {
+    @Test
+    public void givenAString_whenWrittenToFileInputStream_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        List<String> lines = Arrays.asList("Hello. This is just a test. Good bye.");
+        Files.write(sampleOut, lines);
+        File sampleOutFile = sampleOut.toFile();
+        try (InputStream inputStream = new FileInputStream(sampleOutFile)) {
+            Assert.assertTrue(readString(inputStream).contains(lines.get(0)));
+        }
+    }
+
+    @Test
+    public void givenAString_whenWrittenToFileInputStreamWithStringConstructor_thenShouldMatchWhenRead(@TempDir Path tempDir) throws IOException {
+        Path sampleOut = tempDir.resolve("sample-out.txt");
+        String expectedText = "Hello. Hi.";
+        List<String> lines = Arrays.asList(expectedText);
+        Files.write(sampleOut, lines);
+        String fileAbsolutePath = sampleOut.toFile()
+            .getAbsolutePath();
+        try (FileInputStream fis = new FileInputStream(fileAbsolutePath)) {
+            int content;
+            int availabeBytes = fis.available();
+            Assert.assertTrue(availabeBytes > 0);
+            StringBuilder actualReadText = new StringBuilder();
+            while ((content = fis.read()) != -1) {
+                actualReadText.append((char) content);
+            }
```

Comment:
Again, why not reuse the `readString` method here?

Reply:
I wanted to show, in code, an example of some of the methods under FileInputStream.

Reply:
It's identical to the readString logic isn't it? If there is no difference, we should de-duplicate. You can describe the logic in the article with partial snippets etc if you want to.

I agree to keep the availableBytes code to show one of the FileInputStream specific methods, but the rest seems generic.

Reply:
I'll de-duplicate that then and add that logic before section 2.3 in the article.